0.13 docs update: terraform-plugin (#4296)
garden-io/garden: docs/terraform-plugin/about.md

## How it works

Under the hood, Garden simply wraps Terraform, so there's no magic involved. Garden just automates its execution and makes stack outputs available to your Garden providers and actions.

> The `terraform` provider can both provision a Terraform stack when initializing Garden, or through `terraform` actions that are utilized like other actions in your stack.

Reviewer: ```suggestion
Terraform resources can be provisioned through the `terraform` provider when initializing Garden, or via `terraform` actions that are utilized like other actions in your stack.
```

I think this goes smoother

The former, having a single Terraform stack for your whole project, is most helpful if other provider configurations need to reference the outputs from your Terraform stack, or if most/all of your services depend on the infrastructure provisioned in your Terraform stack. A good example of this is the [terraform-gke example](https://github.com/garden-io/garden/tree/0.12.51/examples/terraform-gke) project, which provisions a GKE cluster that the `kubernetes` provider then runs on, along with the services in the project. The drawback is that Garden doesn't currently watch for changes in those Terraform files, and you need to restart to apply new changes, or apply them manually.

> The latter method, using one or more `terraform` _Deploy actions_, can be better if your other providers don't need to reference the stack outputs but your other Deploy, Run and Test actions do. In this style, you can basically create small Terraform stacks that are part of your Stack Graph much like other services. A good example would be deploying a database instance, that other services in your project can then connect to.

Reviewer: ```suggestion
Using `terraform` _Deploy actions_, can be better if your other providers don't need to reference the stack outputs but other Deploy, Run and Test actions do. In this style, you can basically create small Terraform stacks that are part of your Stack Graph much like other services. A good example would be deploying a database instance, that other services in your project can then connect to.
```

Simplification

To manually plan and apply stacks, we provide the following commands:

```console
garden --env=<env-name> plugins terraform apply-root                     # Runs `terraform apply` for the provider root stack.

> garden --env=<env-name> plugins terraform apply-deploy -- <action-name>  # Runs `terraform apply` for the specified terraform Deploy action.

Reviewer: I think this should be `apply-action` after all, kinda just sounds smoother grammar wise

Author: Sounds good, changed in [d2b2abd](https://github.com/garden-io/garden/pull/4296/commits/d2b2abd24312be9f1ca59fca8d3c0e4d773632ac)
